vault/init.go

```diff
@@ -5,16 +5,41 @@ import (
 	"encoding/hex"
 	"fmt"
 
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/vault/helper/jsonutil"
 	"github.com/hashicorp/vault/helper/pgpkeys"
 	"github.com/hashicorp/vault/shamir"
 )
 
+// unsealMetadataStorageEntry holds metadata about all the unseal key shards.
+// This informaion is stored during the initialization and is fetched post
+// unseal for logging purposes.
+type unsealMetadataStorageEntry struct {
+	// Data is a map from each of the unseal key shard to its respective
+	// identifier. It can either be a UUID or a PGP fingerprint (if PGP keys
+	// were employed to encrypte the shards)
+	Data map[string]*UnsealKeyMetadata `json:"data" structs:"data" mapstructure:"data"`
+}
+
+// UnsealKeyMetadata holds metadata associated with each unseal key shard
+type UnsealKeyMetadata struct {
```

Comment:
I feel like we should allow a user to supply a "name" for each part, so that you can init with like "-n 3 -names=vishal,jeff,armon". So that you can get the ID but the human-readable Name is probably more useful.